collect_app/src/main/java/org/odk/collect/android/map/MapConfigurator.java

```diff
@@ -1,24 +1,27 @@
 package org.odk.collect.android.map;
 
 import android.content.Context;
+import android.content.SharedPreferences;
+import android.content.SharedPreferences.OnSharedPreferenceChangeListener;
 
 import org.odk.collect.android.R;
-import org.odk.collect.android.application.Collect;
 import org.odk.collect.android.map.WmsBaseLayerSource.WmsOption;
 import org.odk.collect.android.preferences.GeneralKeys;
 import org.odk.collect.android.preferences.PrefUtils;
-import org.odk.collect.android.spatial.TileSourceFactory;
 
-import java.io.File;
+import java.util.Map;
+import java.util.WeakHashMap;
```

Comment:
Naming: the `MapConfigurator` doesn't really configure the maps anymore—it does still define the Base Layer menus, but it doesn't actually configure them; it just listens for changes to the preferences, and then hands off the configuration information from the appropriate BaseLayerSource to the MapFragment and says "please apply this configuration to yourself."

So the MapConfigurator is more like a coordination point, and the BaseLayerSources are more like configurators.  Open to ideas for better names!